### PR TITLE
Cleanup on wire protocol process status 

### PR DESCRIPTION
**Pull request by lsebrie**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 5 issues

1. ![CRITICAL][CRITICAL] [RskWireProtocol.java#L49](https://github.com/rsksmart/rskj/blob/0caa1ba8146d0a9cd1dcb9c9b2836c6a262f915b/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L49): Unused import - java.math.BigInteger. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![MAJOR][MAJOR] [RskWireProtocol.java#L163](https://github.com/rsksmart/rskj/blob/0caa1ba8146d0a9cd1dcb9c9b2836c6a262f915b/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L163): Method co.rsk.net.eth.RskWireProtocol.processStatus(StatusMessage, ChannelHandlerContext) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [RskWireProtocol.java#L174](https://github.com/rsksmart/rskj/blob/0caa1ba8146d0a9cd1dcb9c9b2836c6a262f915b/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L174): Method co.rsk.net.eth.RskWireProtocol.processStatus(StatusMessage, ChannelHandlerContext) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [RskWireProtocol.java#L185](https://github.com/rsksmart/rskj/blob/0caa1ba8146d0a9cd1dcb9c9b2836c6a262f915b/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L185): Method co.rsk.net.eth.RskWireProtocol.processStatus(StatusMessage, ChannelHandlerContext) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MINOR][MINOR] [RskWireProtocol.java#L246](https://github.com/rsksmart/rskj/blob/0caa1ba8146d0a9cd1dcb9c9b2836c6a262f915b/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L246): Rename "genesis" which hides the field declared at line 81. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java [hunk 1]
```diff
@@ -151,29 +156,37 @@ public void channelRead0(final ChannelHandlerContext ctx, EthMessage msg) throws
...
+                loggerNet.info("Different network received: config network ID {} - message network ID {}",
+                               config.networkId(), msg.getNetworkId());
                 ethState = EthState.STATUS_FAILED;
                 recordEvent(EventType.INVALID_NETWORK);
                 disconnect(ReasonCode.NULL_IDENTITY);
+                ctx.pipeline().remove(this);
+                return;
+            }
+
+            if (!Arrays.equals(msg.getGenesisHash(), genesis.getHash().getBytes())) {
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java]: Do you think you could make `getGenesisHash` return a `Keccak256` instance and do `genesis.getHash().equals(msg.getGenesisHash())`? You would also avoid `new`ing in the log below.

Comment (adauto82) [reply rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java]: I think i kind of fixed this.

**Comment by rskops** [comment]
SonarQube analysis reported 1 issue

1. ![CRITICAL][CRITICAL] [RskWireProtocol.java#L52](https://github.com/rsksmart/rskj/blob/a525315ff34b36cbd50f4af9cc8d2951cde94a84/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L52): Unused import - java.util.Arrays. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by tinchou** [comment]
@adauto82 why didn't you reuse the `genesis` field?

File: rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java [hunk 2]
```diff
@@ -151,29 +154,44 @@ public void channelRead0(final ChannelHandlerContext ctx, EthMessage msg) throws
...
+            int networkId = config.networkId();
+            int msgNetworkId = msg.getNetworkId();
+            if (msgNetworkId != networkId) {
+                loggerNet.info("Removing EthHandler for {} due to invalid network", ctx.channel().remoteAddress());
+                loggerNet.info("Different network received: config network ID {} - message network ID {}",
+                        networkId, msgNetworkId);
                 ethState = EthState.STATUS_FAILED;
                 recordEvent(EventType.INVALID_NETWORK);
                 disconnect(ReasonCode.NULL_IDENTITY);
+                ctx.pipeline().remove(this);
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java]: Could you split this addition as a separate commit? In the future, if anything weird occurs b/c of this line we can track this change and at least know that it was a conscious change and not just a missing line between a bigger code reordering

Comment (lsebrie) [reply rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java]: Ok!

**Comment by rskops** [comment]
SonarQube analysis reported 5 issues

1. ![CRITICAL][CRITICAL] [RskWireProtocol.java#L49](https://github.com/rsksmart/rskj/blob/93c3186c09f145ba9ed085f1af6a5be2f25d4e4a/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L49): Unused import - java.math.BigInteger. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![MAJOR][MAJOR] [RskWireProtocol.java#L163](https://github.com/rsksmart/rskj/blob/93c3186c09f145ba9ed085f1af6a5be2f25d4e4a/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L163): Method co.rsk.net.eth.RskWireProtocol.processStatus(StatusMessage, ChannelHandlerContext) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [RskWireProtocol.java#L174](https://github.com/rsksmart/rskj/blob/93c3186c09f145ba9ed085f1af6a5be2f25d4e4a/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L174): Method co.rsk.net.eth.RskWireProtocol.processStatus(StatusMessage, ChannelHandlerContext) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [RskWireProtocol.java#L185](https://github.com/rsksmart/rskj/blob/93c3186c09f145ba9ed085f1af6a5be2f25d4e4a/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L185): Method co.rsk.net.eth.RskWireProtocol.processStatus(StatusMessage, ChannelHandlerContext) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MINOR][MINOR] [RskWireProtocol.java#L246](https://github.com/rsksmart/rskj/blob/93c3186c09f145ba9ed085f1af6a5be2f25d4e4a/rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java#L246): Rename "genesis" which hides the field declared at line 81. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**